src/constant.h

```diff
@@ -7,9 +7,12 @@
 // Global constants used by Moria
 // clang-format off
 
-#define MAX_UCHAR         255
-#define MAX_SHORT       32767   // maximum short/long signed ints
-#define MAX_LONG   0x7FFFFFFFL
+#include <limits>
+#include <cstdint>
```

Comment:
Eventually I'll probably stop just dumping all includes in the `headers.h`, but perhaps for the moment we can put these there?